src/coreclr/vm/gchelpers.cpp

```diff
@@ -410,7 +410,7 @@ OBJECTREF AllocateSzArray(MethodTable* pArrayMT, INT32 cElements, GC_ALLOC_FLAGS
     }
 #endif
 
-    if (totalSize >= g_pConfig->GetGCLOHThreshold())
+    if (totalSize >= LARGE_OBJECT_SIZE && totalSize >= GCHeapUtilities::GetGCHeap()->GetLOHThreshold())
```

Comment:
Could you please make the same change for NAOT? 

https://github.com/dotnet/runtime/blob/d636522f4ab9615d4358e8bee037ba38ed9e35d1/src/coreclr/nativeaot/Runtime/gcrhenv.cpp#L172